---
permalink: /
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---

I am currently a master student in Quantitative and Systems Biology at Northwestern University (class of '26) after completing my B.A. in Biology at UC Santa Barbara (June 2025). I have received training in both computational methods and molecular experiments. 

Scroll down to see selected results from my projects.


Infer Causal Drivers of Skin Cell Differentiation
======
June 2025 to present

The project integrates variational autoencoder (VAE)-based latent trajectory analysis and Geneformer (a transformer-based algorithm) for single-cell data, and performs in silico perturbations to construct a causal core gene regulatory network that controls human skin cell differentiation.

<img width="1352" height="771" alt="Screenshot 2025-12-01 at 7 45 19 PM" src="https://github.com/user-attachments/assets/e1e931f8-e141-4e60-91ef-3425df8b0277" />

Below is a diagram of the VAE architecture, along with the current results of applying it to identify driver genes for the undifferentiated-to-differentiated transition.

<img width="2560" height="904" alt="image" src="https://github.com/user-attachments/assets/a1c43d1c-9953-4eac-bb59-b48be9f7674b" />
<img width="1321" height="718" alt="Screenshot 2025-12-02 at 12 10 07 AM" src="https://github.com/user-attachments/assets/80afb57c-8811-4c5b-bbc9-45ee6eb0192f" />
<img width="1272" height="375" alt="Screenshot 2025-12-02 at 12 07 19 AM" src="https://github.com/user-attachments/assets/b71185bb-0ed5-4026-bbe6-83565ca8adb7" />

The UMAP demonstrates using VAE to cluster genes by function in an unsupervised manner.

<img width="1313" height="751" alt="Screenshot 2025-12-02 at 12 15 47 AM" src="https://github.com/user-attachments/assets/8e9f76b4-eb4b-424e-abcf-db9b36ae601b" />


Apply Control Theory to Redirect Cell Fate Transition 
======
Aug 2025 to present (remote)

Using RNA velocity Jacobians reconstructed by Dynamo (PMID 35108499), this project applies iLQR (an iterative optimal-control algorithm) to a system where genes serve as control channels, yielding minimal perturbation schedules that steer cells from one fate to another.

Below is the UMAP projection of RNA velocity from the hematopoietic dataset. The red line shows the reference Least Action Path computed by the Dynamo package, and the black line shows the actual trajectory generated by controlling gene expression with the intensity and timing specified in the accompanying Gantt plot.

From Erythroid lineage to Hematopoetic Stem Cells by upregulation of MYCN and GFI1 (both are stem cell maintaince factor. see PMID 15545632 / 15457180).

<img width="924" height="605" alt="gjhkhj" src="https://github.com/user-attachments/assets/c82a2b7e-7545-4574-b058-b469c059d259" />
<img width="2195" height="460" alt="chjkghj" src="https://github.com/user-attachments/assets/1b1c81c2-a383-4069-b4ef-5e000f5acebf" />

From Erythoid lineage to Megakaryocyte by inhibiting ALAS2 (key TF for Ery, see PMID 19656490) and upregulating RGS18 (key TF for Meg, see PMID 25405900).

<img width="791" height="565" alt="chbjhjh" src="https://github.com/user-attachments/assets/61db18aa-873d-4ba3-9d06-39df3a2848e8" />
<img width="2179" height="460" alt="hgjkhg" src="https://github.com/user-attachments/assets/51b9e588-7ead-46b4-8e85-b7bd8733c169" />

and control with 1 gene:

<img width="924" height="605" alt="dhgjaskk" src="https://github.com/user-attachments/assets/1e12e229-5322-4d9c-b37d-0e6ce6dddeb3" />
<img width="2174" height="372" alt="sadjhkl" src="https://github.com/user-attachments/assets/66c8c8b6-7ee1-400c-bcd9-ffcff604b392" />

The system dynamics follow:

$$x_{k+1} = x_k + \Delta t \cdot [f_{vel}(x_k) + \Delta \cdot u_k]$$
- x_k: PCA state vector (30D), representing the state of the system at at time k
- f_vel: velocity field
- Δ: gene PCA loadings
- u_k: gene perturbations (control input)

LQR problem with cost function:

$$ J = Σ_k [x_k^T Q_k x_k + u_k^T R_k u_k] + x_K^T Q_f x_K $$
- Q_k: State deviation cost matrix
- R_k: Control strength cost matrix
- Q_f: Distance to terminal cost matrix

Optogenetic & Cellular Reporter Systems for Reprogramming-induced Rejuvenation
======
Dec 2022 – May 2025

I engineered synthetic gene circuits enabling blue-light-inducible OSK (epigenetic reprogramming factors used to induce epigenetic rejuvenation) activation using the OptoTet system (See PMID 30304687).

Optogenetic Experiments:

<img width="2322" height="1246" alt="image" src="https://github.com/user-attachments/assets/d75b60d4-59bf-432c-a9dc-1fc85d6f367d" />

The system also has a non-optogenetic, Dox-inducible version. Upon induction, OSK along with mCherry is expressed. See the IF staining below.
<img width="686" height="332" alt="Screenshot 2025-12-02 at 9 12 20 PM" src="https://github.com/user-attachments/assets/26e5f6f4-ce0d-4846-aca6-cc49d08cc6df" />

Below are the LED illumination scheme, the corresponding mCherry-channel image, and its quantification.

<img width="769" height="306" alt="Screenshot 2025-12-02 at 12 18 48 AM" src="https://github.com/user-attachments/assets/66854e71-c0e3-4943-a551-00dfff3edbe7" />
<img width="853" height="422" alt="Screenshot 2025-12-02 at 12 18 59 AM" src="https://github.com/user-attachments/assets/ec13a141-b469-4ddc-82b0-6751636f22a0" />
<img width="292" height="205" alt="Screenshot 2025-12-02 at 12 19 10 AM" src="https://github.com/user-attachments/assets/9332fea1-c55f-4959-a433-c6f90157af85" />

I also constructed eight OSK-responsive cellular reporters to enable high-throughput screening for OSK-activatable anti-aging compounds.

<img width="1101" height="832" alt="Screenshot 2025-12-02 at 9 22 24 PM" src="https://github.com/user-attachments/assets/4a03301c-99fc-4c17-b9d0-693fb65f134f" />


I also assisted with other projects in the lab. For example, I reengineered a stress-reporter cell line that was prone to signal saturation by incorporating a small molecule to enable controlled signal degradation, and the graph below shows its time-lapse performance after stress is induced.

<img width="567" height="341" alt="image" src="https://github.com/user-attachments/assets/df5c4ba5-47c4-4142-96b8-93c1f7456a67" />


Building ODE Models of Stress-Induced Cellular Fate Decisions
======
May 2025 – Jun 2025

I applied the GeneSurrounder algorithm (PMID 31060502) to fibroblast transcriptomes to identify integrated stress response regulators as candidate senescence drivers, then built a minimal ODE model linking stress intensity and duration to hormesis, senescence and apoptosis outcomes.

<img width="1824" height="558" alt="image" src="https://github.com/user-attachments/assets/f96033f0-2e4a-4ebd-a6fa-a3aea728d143" />

$$ F(S) = \frac{S^n}{K_S^n + S^n} $$

$$ \frac{d\,\mathrm{PKR}}{dt} = \alpha_1 \cdot \mathrm{pPKR} - \eta \cdot F(S) \cdot \mathrm{PKR} + \alpha_0 \cdot \mathrm{EIF2a} - \alpha_0 \cdot \mathrm{PKR} $$
$$ \frac{d\,\mathrm{pPKR}}{dt} = \eta \cdot F(S) \cdot \mathrm{PKR} - \alpha_1 \cdot \mathrm{pPKR} - \delta_0 \cdot \mathrm{pPKR} $$

$$ \frac{d\,\mathrm{EIF2a}}{dt} = \frac{\beta_1 \cdot (\mathrm{pEIF2a} \cdot \mathrm{GADD34})^2}{K_1 + \mathrm{GADD34}^2} - \alpha_2 \cdot \mathrm{pPKR} \cdot \mathrm{EIF2a} $$ 

$$ \frac{d\,\mathrm{pEIF2a}}{dt} = \alpha_2 \cdot \mathrm{pPKR} \cdot \mathrm{EIF2a} - \frac{\beta_1 \cdot (\mathrm{pEIF2a} \cdot \mathrm{GADD34})^2}{K_1 + \mathrm{GADD34}^2} $$

$$ \frac{d\,\mathrm{GADD34}}{dt} = \frac{\kappa \cdot \mathrm{pEIF2a}^2}{K_2 + \mathrm{pEIF2a}^2} - \beta_2 \cdot \mathrm{GADD34} $$

$$ \frac{d\,\mathrm{p21}}{dt} = \gamma_{\mathrm{p21}} \cdot \mathrm{GADD34} - \delta_{\mathrm{p21}} \cdot \mathrm{p21} $$

$$ \frac{d\,\mathrm{CHOP}}{dt} = \frac{\gamma \cdot \mathrm{pEIF2a}^2}{K_3 + \mathrm{pEIF2a}^2} - \delta_{\mathrm{CHOP}} \cdot \mathrm{CHOP} $$

$$ \frac{d\,\mathrm{DR5}}{dt} = \frac{\gamma_{\mathrm{DR5}} \cdot \mathrm{CHOP}^2}{K_4 + \mathrm{CHOP}^2} - \delta_{\mathrm{DR5}} \cdot \mathrm{DR5} $$

Below is a 3D cube of cell fate. X-axis: stress duration; Y-axis: time after stress onset; Z-axis: stress magnitude. Red: apoptosis. Blue: Senescence. Green: Hormesis

<img width="1314" height="1344" alt="image" src="https://github.com/user-attachments/assets/69e001b1-6e10-485a-9135-5412361b7ad7" />


Reconstructing 3D Waddington Landscapes from RNA velocity
======
Mar 2024 – Apr 2024

The project applied field-decomposition to turn RNA velocity vector field computed from published single cell datasets (PMID 36229609) into 3D Waddington landscape describing the epigenetic potential for cell fate decisions.

<img width="1186" height="547" alt="Screenshot 2025-12-02 at 12 24 07 AM" src="https://github.com/user-attachments/assets/cfc6dc61-1213-4577-ba53-da91604d7499" />


